src/functions.js

```diff
@@ -25,144 +41,113 @@ function setFunctionType(block, ...types) {
+  buildFuctionBlockWith(name, functionType, block => {
+    block.appendValueInput("LEFT")
+    block.appendValueInput("RIGHT")
+      .appendField(field)
+  })
+
+const newListType = (elementType) => new ListType(createType(elementType))
+
+buildFuctionBlock("even", ["Number", "Boolean"])
+buildFuctionBlock("not", ["Boolean", "Boolean"])
+buildFuctionBlock("length", ["String", "Number"])//TODO: List(Char)
+buildFuctionBlock("charAt", ["Number", "String", "String"])
+
+buildInfixFuctionBlock(["compare", ">"], ["a", "a", "Boolean"])//TODO: Selector
+buildInfixFuctionBlock(["apply", "$"], [["a", "b"], "a", "b"])
+
+buildFuctionBlock("id", ["a", "a"])
+buildFuctionBlock("composition", [["b", "c"], ["a", "b"], "a", "c"], ["", ".", "$"])
+
+buildInfixFuctionBlock(["at", "!!"], [newListType("a"), "Number", "a"])
+buildFuctionBlock("any", [["a", "Boolean"], newListType("a"), "Boolean"])
+buildFuctionBlock("all", [["a", "Boolean"], newListType("a"), "Boolean"])
+buildFuctionBlock("filter", [["a", "Boolean"], newListType("a"), newListType("a")])
+buildFuctionBlock("map", [["a", "b"], newListType("a"), newListType("b")])
+buildFuctionBlock("maximum", [newListType("a"), "a"])
+buildFuctionBlock("minimum", [newListType("a"), "a"])
+buildFuctionBlock("fold", [["a", "b", "a"], "a", newListType("b"), "a"])
+
+Blockly.Blocks['list'] = {
   init: function () {
-    this.appendValueInput("LEFT")
-
-    this.appendValueInput("RIGHT")
-      .appendField(">");//TODO: Selector
-
-    this.setInputsInline(true);
-    this.setOutput(true);
-    this.setColour(230);
-    this.setTooltip("");
-    this.setHelpUrl("");
-    this.setOnChange(onChangeFunction.bind(this))
-
-    //Parametric Type
-    setFunctionType(this, "a", "a", "Boolean")
-  }
+    this.appendValueInput("ELEMENT")
+      .appendField("[")
+      .inputType = createType("a")
+    this.appendDummyInput("CLOSE")
+      .appendField("]")
+    this.setInputsInline(true)
+    this.setOutput(true, null)
+    this.setTooltip("")
+    this.setHelpUrl("")
+    this.setOnChange(function (event) { onChangeList.bind(this)(event) })
+    this.outputType = new ListType(createType("a"))
+  },
+
+  /**
+   * Create XML to represent the (non-editable) name and arguments.
+   * @return {!Element} XML storage element.
+   * @this Blockly.Block
+   */
+  mutationToDom: function () {
+    var container = document.createElement('mutation')
+    this.inputList.filter(isBlockInput).forEach(input => {
+      var parameter = document.createElement('arg')
+      parameter.setAttribute('name', input.name)
+      container.appendChild(parameter)
+    })
+    return container
+  },
+
+  /**
+   * Parse XML to restore the (non-editable) name and parameters.
+   * @param {!Element} xmlElement XML storage element.
+   * @this Blockly.Block
+   */
+  domToMutation: function (xmlElement) {
+    this.inputList.forEach(removeInput(this))
+
+    for (var i = 0, childNode; childNode = xmlElement.childNodes[i]; i++) {
```

Comment:
I think this can be a filter and a  forEach, right? :thinking:

Reply:
Ahora no estoy seguro, pero creo que tenía problemas y por eso quedó así.